python/triton/experimental/gluon/language/amd/gfx1250/__init__.py

```diff
@@ -15,3 +19,44 @@ def wmma(a, b, acc, _semantic=None):
         acc (tensor): The accumulator tensor.
     """
     return _wmma(3, a, b, acc, _semantic)
+
+
+@builtin
+def wmma_scaled(a, a_scale, a_format, b, b_scale, b_format, acc, _semantic=None):
+    """
+    AMD Scaled WMMA operation.
+
+    ```
+    c = a * a_scale @ b * b_scale + acc
+    ```
+
+    `a` and `b` use microscaling formats described in
+    "OCP Microscaling Formats (MX) Specification":
+    https://www.opencompute.org/documents/ocp-microscaling-formats-mx-v1-0-spec-final-pdf.
+
+    Args:
+        a (tensor): The operand A to be multiplied.
+        a_scale (tensor): Scale factor for operand A.
+        a_format (str): Format of the operand A. Available formats: `e2m1'.
+        b (tensor): The operand B to be multiplied.
+        b_scale (tensor): Scale factor for operand B.
+        b_format (str): Format of the operand B. Available formats: `e2m1'.
+        acc (tensor): Accumulator tensor.
+    """
+    _verify_wmma(3, a, b, acc)
+    if a_format.value == "e2m1":
+        wmma_layout = a.type.layout.parent
+        assert isinstance(wmma_layout, AMDWMMALayout) and wmma_layout.instr_shape == (16, 16, 64), \
+            "e2m1 format expects instr_shape to be (16, 16, 64)"
+    if b_format.value == "e2m1":
+        wmma_layout = b.type.layout.parent
+        assert isinstance(wmma_layout, AMDWMMALayout) and wmma_layout.instr_shape == (16, 16, 64), \
+            "e2m1 format expects instr_shape to be (16, 16, 64)"
+
```

Comment:
Can we add verification to check that `acc`'s WMMA instr shape is `(16, 16, 128)`? It's very easy to assume that they need to match otherwise and head scratching why things doesn't work..